packages/axios-extension/src/factory.ts

```diff
@@ -41,6 +41,14 @@ function createInstance<T extends ServiceProvider>(
     delete providerConfig.ignoreCertErrors;
     providerConfig.withCredentials = providerConfig?.auth && Object.keys(providerConfig.auth).length > 0;
 
+    /**
+     * Make axios throw an error for 4xx errors as well.
+     *
+     * @param status - http response status
+     * @returns success (true) or error (false)
+     */
+    providerConfig.validateStatus = (status) => status < 400;
```

Comment:
Where is this used? Also 3xx codes (redirection) need the client to take additional steps, so getting a 3xx code does not indicate success. Do we not expect 3xx codes?

Reply:
We don't have any special handling for `3xx` in our code, and as far as I can see, all the redirects are handled by `axios` transparently except you disable it using `maxRedirects: 0`.
By default `axios` only considers `>=500` as errors which lead to many parts in the code where we check a successful response for a `4xx` response and if we find it, we throw an error. This line makes `axios` to also throw `4xx` as errors right away which fits the usage of our library.